### PR TITLE
Fix failure to convert string "uint64" to NaN

### LINKED ISSUE #32394: pd.to_numeric(..., errors="coerce") failing silently when strings contain "uint64"
#### Problem description
When trying to coerce strings to numeric values using `to_numeric()`, the occurrence of the substring "uint64" (but not any other dtype-like substring it seems) leads to silent failure to coerce.

```python
strs = ["32", "64", "uint32", "float64", "sdnfonsdf uint32 knsdf", "sdnfonsdf uint64 knsdf", "uint64"]
print([pd.to_numeric(s, errors="coerce") for s in strs])
pd.to_numeric(pd.Series(["32", "64", "uint64"]), errors="coerce")
```

```bash
[32, 64, nan, nan, nan, 'sdnfonsdf uint64 knsdf', 'uint64']

0        32
1        64
2    uint64
dtype: object
```

#### Expected Output
```bash
[32, 64, nan, nan, nan, nan, nan]

0        32.0
1        64.0
2        NaN
dtype: float64
```

Seems to fail equally in 0.25.3 and 1.0...

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None

pandas           : 0.25.3
numpy            : 1.17.3
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 45.1.0.post20200119
Cython           : None
pytest           : 5.3.4
hypothesis       : None
sphinx           : 2.3.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.4.1
html5lib         : None
pymysql          : 0.9.3
psycopg2         : 2.8.4 (dt dec pq3 ext lo64)
jinja2           : 2.10.3
IPython          : 7.11.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.4.1
matplotlib       : 3.1.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.14.1
pytables         : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.12
tables           : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by roberthdevries**
Including regression test

- [x] closes #32394
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm - @gfyoung care to look?

**Review by gfyoung — APPROVED** [review]
Nice!

The fix I had back then was good, but I guess not all patches age well...

**Comment by WillAyd** [comment]
Great thanks @roberthdevries - keep the PRs coming!

File: pandas/tests/dtypes/test_inference.py [hunk 1]
```diff
@@ -507,6 +507,13 @@ def test_convert_numeric_int64_uint64(self, case, coerce):
         result = lib.maybe_convert_numeric(case, set(), coerce_numeric=coerce)
         tm.assert_almost_equal(result, expected)
 
+    def test_convert_numeric_string_uint64(self):
+        # GH32394
+        result = lib.maybe_convert_numeric(
+            np.array(["uint64"], dtype=object), set(), coerce_numeric=True
+        )
+        assert np.isnan(result)
```
Comment (jreback) [inline pandas/tests/dtypes/test_inference.py]: this is not a good test - does it replicate what the OP was? this is not user facing at all

pls follow what is going on in other tests

**Comment by jreback** [comment]
@WillAyd this has a number of issues in testing

pls rever

File: pandas/tests/dtypes/test_inference.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/tests/dtypes/test_inference.py]: The OP currently takes this path, though it'd be good to test at the `pd.to_numeric` level too.

**Review by TomAugspurger — COMMENTED** [review]
If the issue is just lack of tests I think we can push that in an additional PR, rather than reverting.

**Comment by roberthdevries** [comment]
@jreback I will add another PR with a test for to_numeric referencing the original issue

**Pull request merged**